### ISSUE TITLE
Confusion over nested relationships and links url

### ISSUE THREAD

**Issue by devpascoe** [json-api/json-api#550]
Hey team, trying to get my head around this latest spec.
So i have profile and location tables. A profile has a many-many to itself (so profiles can follow other profiles).
And so i want to do just that, fetch profiles i'm following passing my profile id.
If i call my imaginary endpoints with /profiles/1 i should receive a compound document of the profile with a "following" attribute within its links. And included will have the profiles followed.

```
{
    "data": {
        "created_at": "2015-04-12T23:51:19Z",
        "id": 5,
        "links": {
            "following": {
                "linkage": [
                    {
                        "id": 3,
                        "type": "profiles"
                    }
                ]
            },
            "self": "/profiles/5",
            "user": {
                "linkage": {
                    "id": 2,
                    "type": "users"
                }
            }
        },
        "profile_type": "user",
        "type": "profiles",
        "updated_at": "2015-04-12T23:51:19Z"
    },
    "included": [
        {
            "created_at": "2015-04-09T07:16:55Z",
            "id": 3,
            "profile_type": "location",
            "type": "profiles",
            "updated_at": "2015-04-09T07:16:55Z"
        },
        {
            "created_at": "2015-04-12T23:51:19Z",
            "id": 2,
            "type": "users",
            "updated_at": "2015-04-12T23:51:19Z",
            "user_id": 1
        }
    ],
    "links": {
        "self": "/profiles/5"
    }
}
```

However, i need to get each profile's one-one related location data. I would love to add it in the included profiles however i'm thinking this breaks the spec.... (notice my inclusion of the nested related attributes for location)

```
        {
            "created_at": "2015-04-09T07:16:55Z",
            "id": 3,
            "profile_type": "location",
            "type": "profiles",
            "updated_at": "2015-04-09T07:16:55Z",
            "location": {
                "id": 1,
                "title": "test location"
            }
        }
```

Any tips to achieving this (preferably within a single call) would be epic. Thanks.

**Comment by devpascoe** [comment]
I know i could call something like /profiles/5/links/following and get back something like

```
{
  "links": {
    "self": "/profiles/5/links/following",
    "related": "/profiles/5/following"
  },
  "data": [
    { "type": "profile", "id": "2" },
    { "type": "profile", "id": "3" }
  ]
}
```

... but again, how do i get each related profile's location attributes included in this single call

side question, can i have an included attribute on these calls? i really need to get more data out of my relationships

**Comment by ethanresnick** [comment]
Eventually, I think you should be able to do something like: `GET /profiles/1?include=following,following.location`. That will include the linked `following` profiles, and the `location` resource that each of those included profiles links to. 

But the exact format for this response (i.e. which resources/relationship names will appear where in the response document) is still being worked out in #497. This is, imo, one of the one or two issues that still needs to be resolved before 1.0.

**Comment by tkellen** [comment]
Closing this in favor of #497 (and a pending PR to resolve). I hear your concerns @devpascoe and we are actively working to reach clarity on this point in the specification.
